src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -4529,8 +4588,7 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
                                     (control == static_cast<uint8_t>(~0xAA & 0xF0)) || // ~C & A
                                     (control == static_cast<uint8_t>(~0xAA & 0xCC)))   // ~C & B
                                 {
-                                    // We're normalizing to ~B & C, so we need another swap
-                                    std::swap(*val2, *val3);
+                                    // We already normalized to ~B & C above.
```

Comment:
nit: This first if could be merged with the assert below. We may not even need the assert at all since it's the only possible 6 permutations of `NOT AND` and `NOT OR`

Reply:
I kept the logic here stylized to be the same as the hoisted logic, for better or worse...